service/history/workflowcache/cache.go

```diff
@@ -135,12 +136,20 @@ func (c *wfCache) allow(domainID string, workflowID string, rateLimitType rateLi
 
 	switch rateLimitType {
 	case external:
+		if !c.workflowIDCacheEnabled(domainName) && !c.workflowIDCacheExternalEnabled(domainName) {
```

Comment:
why do we need both `workflowIDCacheEnabled` and `workflowIDCacheExternalEnabled`?

Reply:
I had kept it because workflowIDCacheEnabled was already in use within Uber internal. I have removed it after checking with @jakobht that the usage is minimal and we can replace it